integration_tests/pages/match/placementApplicationWithdrawalConfirmationPage.ts

```diff
@@ -1,11 +1,27 @@
+import { ApprovedPremisesUserRole } from '@approved-premises/api'
 import { WithdrawPlacementRequestReason } from '../../../server/@types/shared/models/WithdrawPlacementRequestReason'
 import Page from '../page'
+import {
+  noCapacityReasons,
+  placementNoLongerNeededReasons,
+  problemInPlacementReasons,
+} from '../../../server/utils/applications/withdrawables/withdrawalReasons'
 
 export default class ConfirmationPage extends Page {
   constructor() {
     super('Why is this request for placement being withdrawn?')
   }
 
+  shouldShowReasons(userRoles: Array<ApprovedPremisesUserRole>) {
+    const reasons = userRoles.includes('workflow_manager')
+      ? [placementNoLongerNeededReasons, noCapacityReasons, problemInPlacementReasons].flat()
+      : [placementNoLongerNeededReasons, problemInPlacementReasons].flat()
+
+    cy.get('.govuk-radios__input').each((inputElement, inputElementIndex) =>
+      cy.wrap(inputElement).should('have.value', reasons[inputElementIndex]),
+    )
+  }
```

Comment:
hmm, I wonder if we could be a bit clearer here? e.g.

1) be explicit in the name of this assertion that we are expecting the list of available reasons to vary depending on the user's role e.g. `shouldShowReasonsApplicableToRole()`, or

2) have two separate assertion functions named explicitly for the conditional logic in play e.g.

- `shouldShowReasonsAvailableToAllRoles()` and 
- `shouldShowReasonsAvailableToWorkflowManagerRoleOnly()`

What do you think / prefer?

Reply:
So the difficulty here is that the radio options end up in a flat array and the CRU-specific ones are in the middle, hence the use of index-matching in the assertions. It might be worth extending the radio options generator to allow a test ID attribute for easier selection, which should allow us to assert on each of the reason types individually

For now though, I've split it into two assertion functions: one for all reasons and one for non-capacity-related reasons. These could be renamed to refer to the role instead of the reason type, but I opted for the latter because the index matching approach would prevent `shouldShowReasonsAvailableToAllRoles` from working for workflow managers, which might be a little confusing given it sounds as though should apply to all roles, and `nonWorkflowManager` feels awkward

What do you think?

Fixup commit showing the change: https://github.com/ministryofjustice/hmpps-approved-premises-ui/pull/1562/commits/e1e2237f4302c399c7d2681d7e99bb17ca50d201

Reply:
(Could just as easily go with `shouldShowReasonsApplicableToRole` though!)